synapse/handlers/room_summary.py

```diff
@@ -93,7 +94,9 @@ def __init__(self, hs: "HomeServer"):
         self._event_serializer = hs.get_event_client_serializer()
         self._server_name = hs.hostname
         self._federation_client = hs.get_federation_client()
-        self._request_ratelimiter = hs.get_request_ratelimiter()
+        self._ratelimiter = Ratelimiter(
+            store=self._store, clock=hs.get_clock(), rate_hz=5, burst_count=10
```

Comment:
Where did these numbers come from, out of interest?

Reply:
I made them up, they should allow for 5 queries per second without enforcing any ratelimiting until 10 queries. This might be a bit high, but I'm never quite sure what to put here and usually err on the side of allowing more. 🤷

Reply:
Fair enough. We can always tighten the limits!

(Though there's a part of me that wonders if rate limiting ought to be handled by the load balancer to keep Synapse itself simpler.)

Reply:
Some rate limits can be tweaked by the Synapse config, FWIW. They can also depend on particular data, e.g. you could limit the message rate per user, per room, per message type (or something like that).